src/main/java/org/jabref/gui/importer/fetcher/WebSearchPaneViewModel.java

```diff
@@ -130,16 +144,26 @@ public void search() {
         }
 
         SearchBasedFetcher activeFetcher = getSelectedFetcher();
+        Callable<ParserResult> parserResultCallable = () -> new ParserResult(activeFetcher.performSearch(query));
+        String fetcherName = activeFetcher.getName();
+
+        Optional<DOI> doi = DOI.findInText(query);
+        if (doi.isPresent()) {
+            DoiFetcher doiFetcher = new DoiFetcher(preferencesService.getImportFormatPreferences());
+            parserResultCallable = () -> new ParserResult(OptionalUtil.toList(doiFetcher.performSearchById(doi.get().getDOI())));
+            fetcherName = doiFetcher.getName();
+        }
+
+        String finalFetcherName = fetcherName;
         Globals.getTelemetryClient().ifPresent(client ->
-                client.trackEvent("search", Map.of("fetcher", activeFetcher.getName()), Map.of()));
+                client.trackEvent("search", Map.of("fetcher", finalFetcherName), Map.of()));
 
-        BackgroundTask<ParserResult> task;
-        task = BackgroundTask.wrap(() -> new ParserResult(activeFetcher.performSearch(query)))
+        BackgroundTask<ParserResult> task = BackgroundTask.wrap(parserResultCallable)
                              .withInitialMessage(Localization.lang("Processing %0", query));
         task.onFailure(dialogService::showErrorDialogAndWait);
 
         ImportEntriesDialog dialog = new ImportEntriesDialog(stateManager.getActiveDatabase().get(), task);
-        dialog.setTitle(activeFetcher.getName());
+        dialog.setTitle(fetcherName);
```

Comment:
The "new" variable should also be used - otherwise, it is confusing IMHO.

```suggestion
        dialog.setTitle(finalFetcherName);
```